### PR TITLE
Fix unused lexical argument warnings

### LINKED ISSUE #345: New byte compiler warnings
When updating dumb-jump to the latest version, I noticed the following byte compiler-warnings:

```
dumb-jump.el:74:1: Warning: defcustom for
    ‘dumb-jump-ivy-jump-to-selected-function’ fails to specify type
dumb-jump.el:74:1: Warning: defcustom for
    ‘dumb-jump-ivy-jump-to-selected-function’ fails to specify type
dumb-jump.el:2053:1: Warning: Unused lexical argument ‘config’

In dumb-jump-handle-results:
dumb-jump.el:2270:153: Warning: reference to free variable ‘rel-cur-file’
dumb-jump.el:2279:1: Warning: Unused lexical argument ‘use-tooltip’
dumb-jump.el:2279:1: Warning: Unused lexical argument ‘look-for’
dumb-jump.el:2390:1: Warning: Unused lexical variable ‘modified-file-buffers’

In dumb-jump-goto-file-line:
dumb-jump.el:2449:17: Warning: ‘find-tag-marker-ring’ is an obsolete variable
    (as of 25.1); use ‘xref-push-marker-stack’ or ‘xref-pop-marker-stack’
    instead.
dumb-jump.el:2765:1: Warning: Unused lexical argument ‘lang’
dumb-jump.el:2785:1: Warning: Unused lexical argument ‘cur-file’
dumb-jump.el:2837:1: Warning: Unused lexical argument ‘exclude-paths’
dumb-jump.el:2837:1: Warning: Unused lexical argument ‘lang’
dumb-jump.el:2942:13: Warning: reference to free variable ‘ctx-type’
dumb-jump.el:2942:93: Warning: reference to free variable ‘prefer-external’
dumb-jump.el:2942:140: Warning: reference to free variable ‘cur-file’
dumb-jump.el:2942:149: Warning: reference to free variable ‘rel-cur-file’
```

### PR DESCRIPTION
**Pull request by thisch**
Partially fixes #345

### CODE REVIEW

File: dumb-jump.el [hunk 1]
```diff
@@ -2465,7 +2465,7 @@ Ffrom the ROOT project CONFIG-FILE."
 
 (defun dumb-jump-file-modified-p (path)
```
Comment (thisch) [inline dumb-jump.el:2466]: This function is IMO broken, since `modified-file-buffers` is not used and AFAICS this function always returns nil.

Comment (jacktasia) [reply dumb-jump.el:2466]: @phikal looks like this got messed up in #330 around [here](https://github.com/jacktasia/dumb-jump/pull/330/files#diff-7d5fde248d17c1592b9a27209496fe24L2353). Would this be easy for you to fix?

Comment (thisch) [reply dumb-jump.el:2466]: Sure. Fixed in #361.

**Comment by thisch** [comment]
Can this be merged?

**Pull request merged**